pageserve/app.py

```diff
@@ -164,13 +164,21 @@ def add_event():
         Redirect to index if 201 response from events service.
         Error message and status 400 otherwise.
     """
-    url = app.config['EVENTS_ENDPOINT'] + 'add'
-    form_data = dict(**request.form.to_dict(), author_id=get_user()['user_id'])
-    r = requests.post(url, data=form_data)
-    if r.status_code == 201:
-        # upload successful, redirect to index
-        return redirect(url_for("index"))
-    return r.content, r.status_code
+    try:
+        user = get_user()
+        if not user['is_organizer']:
+            return 'Error: not authorized to add events.', 403
+        url = app.config['EVENTS_ENDPOINT'] + 'add'
+        form_data = dict(**request.form.to_dict(), author_id=user['user_id'])
+        # get rid of 'T' separator in event_time
+        form_data['event_time'] = form_data['event_time'].replace('T', ' ')
+        r = requests.post(url, data=form_data)
+        if r.status_code == 201:
+            # upload successful, redirect to index
+            return redirect(url_for('index'))
+        return r.content, r.status_code
+    except KeyError as error:
```

Comment:
Should this be BadRequestKeyError instead? Or does it need to handle both?

Reply:
`BadRequestKeyError` won't be thrown since line 172 converts `request.form` to a dict which raises `KeyError`s on missing keys.

Reply:
That works then!